clap_complete/tests/testsuite/dynamic.rs

```diff
@@ -1,34 +1,69 @@
 #![cfg(feature = "unstable-dynamic")]
 
+macro_rules! assert_completion {
+    ($cmd:expr, $input:expr, $(current_dir = $current_dir:expr,)? [$($comp:expr $(=> $help:expr)?),+]) => {
+        let mut args = vec![std::ffi::OsString::from($cmd.get_name())];
+        let arg_index;
+
+        if let Some((prior, after)) = $input.split_once("[TAB]") {
+            args.extend(prior.split_whitespace().map(From::from));
+            if prior.ends_with(char::is_whitespace) {
+                args.push(std::ffi::OsString::default())
+            }
+            arg_index = args.len() - 1;
+            // TODO when we can handle [TAB] inside of tokens, we need to update this
```

Comment:
TODOs in code are another form of backlog that is nearly invisible.

What I would instead do is `HACK: this cannot handle in-word '[TAB]'a`

Reply:
Makes sense, as this is less a Todo to actually implement in word completions but more about testing them when we have them.